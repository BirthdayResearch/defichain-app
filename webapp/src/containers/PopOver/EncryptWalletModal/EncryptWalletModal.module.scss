// App and Bootstrap variables required
@import '~bootstrap/scss/functions';
@import '~bootstrap/scss/variables';
@import '~bootstrap/scss/mixins';
@import '../../../scss/variables';
@import '../../../scss/bootstrap/variables';

.modalContent {
  background-color: #f4f3f6;
  border: none;
}

.modalBackdrop {
  background-color: #f4f3f6;
  opacity: 1 !important;
}

.lockIcon {
  width: 80px;
  height: 80px;
  background-color: $light;
  text-align: center;
  line-height: 80px;
  border-radius: 100%;
  margin-bottom: 20px;
  padding: 20px;
  color: $secondary;
}
