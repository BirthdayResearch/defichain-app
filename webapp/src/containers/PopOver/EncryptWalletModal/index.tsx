import React from 'react';
import { Modal, ModalBody } from 'reactstrap';
import { connect } from 'react-redux';
import styles from './EncryptWalletModal.module.scss';

import {
  closeEncryptWalletModal,
  encryptWalletFailure,
  encryptWalletStart,
} from '../reducer';
import EncryptWalletPage, {
  EncryptWalletPageProps,
} from '../../WalletPage/components/EncryptWalletPage';

interface EncryptWalletModalProps extends EncryptWalletPageProps {
  isEncryptWalletModalOpen: boolean;
  closeEncryptWalletModal: () => void;
}

const EncryptWalletModal: React.FunctionComponent<EncryptWalletModalProps> = (
  props: EncryptWalletModalProps
) => {
  const { isEncryptWalletModalOpen, closeEncryptWalletModal } = props;

  return (
    <Modal
      isOpen={isEncryptWalletModalOpen}
      contentClassName={styles.modalContent}
      backdropClassName={styles.modalBackdrop}
      style={{ width: '100%', maxWidth: '100vw' }}
      centered
    >
      <ModalBody>
        <EncryptWalletPage
          onClose={closeEncryptWalletModal}
          isModal={true}
          pageSize={7}
        />
      </ModalBody>
    </Modal>
  );
};

const mapStateToProps = (state) => {
  const {
    isEncryptWalletModalOpen,
    isWalletEncrypting,
    isErrorEncryptingWallet,
  } = state.popover;
  return {
    isWalletEncrypting,
    isEncryptWalletModalOpen,
    isErrorEncryptingWallet,
  };
};

const mapDispatchToProps = {
  encryptWalletStart,
  encryptWalletFailure,
  closeEncryptWalletModal,
};

export default connect(mapStateToProps, mapDispatchToProps)(EncryptWalletModal);
