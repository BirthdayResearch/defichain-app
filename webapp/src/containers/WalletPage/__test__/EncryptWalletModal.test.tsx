import React from 'react';
import EncryptWalletModal from '../components/EncryptWalletPage/index';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../../../app/rootStore';
import Enzyme from '../../../utils/testUtils/enzyme';

describe('EncryptWalletPage component', () => {
  it('should check for snapshot', () => {
    const wrapper = Enzyme.mount(
      <Router>
        <Provider store={store}></Provider>
      </Router>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
