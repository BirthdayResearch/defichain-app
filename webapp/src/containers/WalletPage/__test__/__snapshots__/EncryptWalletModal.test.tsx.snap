// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EncryptWalletPage component should check for snapshot 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Provider
      store={
        Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        }
      }
    />
  </Router>
</BrowserRouter>
`;
